### ISSUE TITLE
Goals

### ISSUE THREAD

**Issue by jagannathBhat** [hhkerala/discuss#2]
It will be best if we can make it clear what the goals of this project are.

What problems are we solving?
What are the outcomes?
Who are we solving it for?
How can we measure our progress?

@athul

**Comment by jagannathBhat** [comment]
Let's discuss here and create a goals.md in the end

**Comment by athul** [comment]
Sure, maybe @jagannathBhat you could start off with it. Since that'll set some expectations for furthur goals and feedback

**Comment by devpranoy** [comment]
Let's start with the problems first, off the top of my head, a very common problem is that there are no curated hackathon lists for hackers in Kerala we get to know about an event from tech communities.

**Comment by devpranoy** [comment]
Timing is an issue, most hackathons tend to happen when a college semester starts to encourage student participation which is a good thing, but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.

**Comment by devpranoy** [comment]
We had MLH, Beach hack, and Reboot happen within a span of 1 week, so as a hacker I need to prioritize one over the other which leads to a loss of opportunity.

**Comment by athul** [comment]
> Let's start with the problems first, off the top of my head, a very common problem is that there are no curated hackathon lists for hackers in Kerala we get to know about an event from tech communities.

We can make a community curated Hackathon Listing website. We can let the hackers rate hackathons based on their previous experiences and those with better ratings could show up in the top of the site.

>  but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.

That is true. This could be well improved.

**Comment by devpranoy** [comment]
> > Let's start with the problems first, off the top of my head, a very common problem is that there are no curated hackathon lists for hackers in Kerala we get to know about an event from tech communities.
> 
> We can make a community curated Hackathon Listing website. We can let the hackers rate hackathons based on their previous experiences and those with better ratings could show up in the top of the site.
> 
> > but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.
> 
> That is true. This could be well improved.

An IMDB for hackathons is going to be lit

**Comment by athul** [comment]
I had this question

Who should be who?
1. In the Core team
    - Organizers or Hackers
2. The Curation Team
    - Organizers or Hackers

**Comment by athul** [comment]
> How can we measure our progress?

@thetronjohnson suggested a good method here 
https://github.com/hhkerala/discuss/issues/1#issuecomment-586723846

**Comment by de-sh** [comment]
> We can make a community curated Hackathon Listing website. We can let the hackers rate hackathons based on their previous experiences and those with better ratings could show up in the top of the site.

In what ways will we be changing the game, considering that some platforms like [Devfolio](https://devfolio.co) have tried their hand at hosting the online registration process as well as publicity and sponsorship help(marketing? I am not very sure of this) for hackathons(which I hear is not liked by some organisers, but for me as a hacker is working wonderfully).

Considering you have pointed out that it's going to be **community curated**, this might be the difference. Probably we need to make an alternative to their platform and leverage our connections to make it work, within the Kerala hacker circles.

**Comment by icodelifee** [comment]
> Timing is an issue, most hackathons tend to happen when a college semester starts to encourage student participation which is a good thing, but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.

i very much agree on this.. i thought of attending some of the upcoming hackathons recently but all of them are one the same dates of my internals exams.

**Comment by kaiiyer** [comment]
> > Timing is an issue, most hackathons tend to happen when a college semester starts to encourage student participation which is a good thing, but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.
> 
> i very much agree on this.. i thought of attending some of the upcoming hackathons recently but all of them are one the same dates of my internals exams.

Colleges should actually encourage going to hackathons and technical events. There are folks who got selected to Inout and were unable to participate just coz of college administration.
If college does have some sort of tech club and if hackathons organized by them are via this club then fixing the dates so that everyone can attend with ease can be achieved. A decentralized organization( totally Opensource) for managing the dates for hackathon and reaching out to every college club out there can be an effective way.

**Comment by de-sh** [comment]
> Who are we solving it for?

We surely need to focus on the hacker community, i.e. students and professionals from technical backgrounds.

On the other hand we can also provide guid-lines useful to organizers from academia or industry and probably even entrepreneurs who want to build platforms and tools that might come in handy for them.

**Note:** I mean to direct attention at the lack of involvement from innovative entrepreneurs, people should literally try renting equipment to hackers at hackathons!

**Comment by athul** [comment]
> In what ways will we be changing the game, considering that some platforms like Devfolio have tried their hand at hosting the online registration process as well as publicity and sponsorship 

Not like de**** in any way. We don't want to host hackathons there. There are tons of better platforms for it like devpost and others. We only exist in within the kerala Hacker Community and not around India and since this is limited here, we could improve 1-1 communication and reduce the language barrier. Not all of us are fluent in conveying everything in English.  We should exist for the organizers and the Hacker Community. Sponsorships are another totally different Game.

Hell, the hackers in kerala could build something like de**** in a hackathon or something better than that.

**Comment by athul** [comment]
We should be able to **empower organizers** to conduct more standalone hackathons than conducting hackathons as part of their technical fests. We should introduce more people to hackathons and convey its importance. We should be able to help organizers with feedback on how to make your hackathon be the best in Kerala. I'm sorry if I'm being a bit ambitious here, but we should take examples from MLH on how they changed the term Hackathon.

> professionals from technical backgrounds.

Let's not go there yet. Professionals can conduct hackathons as part of their Jobs and all but for students it's not so much easy to organize stuff

**Comment by kaiiyer** [comment]
Actually in most of the hackathons we can see beginners popping up and wanting to start with something but not getting the right idea or vision. I strongly suggest introducing mentors group wise(1 mentor for 3 groups) rather than just spreading out. When we had the mozilla sprint we made groups with a mentor assigned for each of them. Thus the productivity increases and the folks can get much more help than wht they're getting now.

**Comment by icodelifee** [comment]
> > > Timing is an issue, most hackathons tend to happen when a college semester starts to encourage student participation which is a good thing, but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.
> > 
> > 
> > i very much agree on this.. i thought of attending some of the upcoming hackathons recently but all of them are one the same dates of my internals exams.
> 
> Colleges should actually encourage going to hackathons and technical events. There are folks who got selected to Inout and were unable to participate just coz of college administration.
> If college does have some sort of tech club and if hackathons organized by them are via this club then fixing the dates so that everyone can attend with ease can be achieved. A decentralized organization( totally Opensource) for managing the dates for hackathon and reaching out to every college club out there can be an effective way.

first college administration need to go through awareness class about what is hackathons and what are benefits of it. afaik my college doesnt even care about hackathons and will not let us represent the college unless someone won something at these hacks

**Comment by athul** [comment]
> Actually in most of the hackathons we can see beginners popping up and wanting to start with something but not getting the right idea or vision. I strongly suggest introducing mentors group wise(1 mentor for 3 groups) rather than just spreading out. When we had the mozilla sprint we made groups with a mentor assigned for each of them. Thus the productivity increases and the folks can get much more help than wht they're getting now.

This is cool Stuff. Will be adding this in the guidelines.md

**Comment by kaiiyer** [comment]
> > > > Timing is an issue, most hackathons tend to happen when a college semester starts to encourage student participation which is a good thing, but the problem is that the organizers keep their events a day or two apart from one another. Coordination between organizers should be improved.
> > > 
> > > 
> > > i very much agree on this.. i thought of attending some of the upcoming hackathons recently but all of them are one the same dates of my internals exams.
> > 
> > 
> > Colleges should actually encourage going to hackathons and technical events. There are folks who got selected to Inout and were unable to participate just coz of college administration.
> > If college does have some sort of tech club and if hackathons organized by them are via this club then fixing the dates so that everyone can attend with ease can be achieved. A decentralized organization( totally Opensource) for managing the dates for hackathon and reaching out to every college club out there can be an effective way.
> 
> first college administration need to go through awareness class about what is hackathons and what are benefits of it. afaik my college doesnt even care about hackathons and will not let us represent the college unless someone won something at these hacks

Starting a mozilla club/ hackclub or any technical club should be the first initiative and you should choose the faculty who can support technical activities in college. Not only just tech fest but conducting tech events are also part of growing. If there's an initiative like a tech workshop or something the communities can help you out.

**Comment by icodelifee** [comment]
Also i have seen some colleges conducting hacks that rewards like 20 KTU Credits something like that, that should be avoided, because not everyone is a engineering student [myself]

**Comment by athul** [comment]
Personally speaking, the whole idea of the ktu stuff is only put forward if the registrations are low. Most of the hackers won't fall for this and this is just seen as a publicity stunt

**Comment by athul** [comment]
We should continue with this thread

**Comment by jagannathBhat** [comment]
> > How can we measure our progress?
> 
> @thetronjohnson suggested a good method here
> [#1 (comment)](https://github.com/hhkerala/discuss/issues/1#issuecomment-586723846)

Yeah, we need to get user feedback. What I meant was we need to decide what questions should we be asking.

**Comment by jagannathBhat** [comment]
> We should be able to **empower organizers** to conduct more standalone hackathons than conducting hackathons as part of their technical fests. We should introduce more people to hackathons and convey its importance. We should be able to help organizers with feedback on how to make your hackathon be the best in Kerala. I'm sorry if I'm being a bit ambitious here, but we should take examples from MLH on how they changed the term Hackathon.
> 
> > professionals from technical backgrounds.
> 
> Let's not go there yet. Professionals can conduct hackathons as part of their Jobs and all but for students it's not so much easy to organize stuff

I guess the problem is, organizers shortlist people based on their abilities and select people with more experience in developing projects. Instead, organizers should focus on selecting people with less experience or beginners and organizing workshops along with the hackathon to train them.

**Comment by jagannathBhat** [comment]
From what I've read in this thread I've derived the following goals.
1. Show a list of hackathons happening in Kerala.
2. Help organizers schedule hackathons in a way that the dates don't clash with each other or exam dates.
3. IMDB for hackathons. Let participants rate hackathons and organizers.
4. Help participants find mentors prior to hackathons.

Have I missed anything?

**Comment by thetronjohnson** [comment]
https://forms.zohopublic.in/kiranjohns369/form/HackathonHackersKerala/formperma/ZtNHAc7zO4uMYy1tjrwi_1NoYHOWVN5crUv-tbOkvE8

Made a survey form! If we could get to know what participants expect and what problems organizers face, get some info from their end we'll get an idea on how to orient this project

Anything else to add in this?

**Comment by athul** [comment]
@gauthamzz gave very Valuable feedback on hackathons via chat 
> Most of the things I believe in I wrote in the blog

>We never put any faculty as a judge for hackathon, I'm seeing you have one. Idk if it's good. People 
 ended up always hating it.

>And it's always best to start early and get sponsors then decide prizes later

>We used to do hackathon in March,We used to send sponsorship mails from September.Then only you 
 can get it. It's like one year long

>And sponsorship is the most important thing

>I normally don't like hackathon winners and stuff. I like giving too 5 same amount. So it's more of a 
 everyone wins thing.

>And you need to ensure everyone gets at-least a t-shirt or no one will like the event

>It's like making everyone feel like they build something cool

>And to get sponsors just look what other hackathons are getting as sponsors then email them
  We used to look at hackathons in US, since at that time we didn't have any other hackathons

**Comment by de-sh** [comment]
> I normally don't like hackathon winners and stuff. I like giving too 5 same amount. So it's more of a
everyone wins thing.

Well said!
I also believe that this needs more spotlight atm. Too many hackathons are becoming competitions, places where there is a lack of collaboration, while they are really meant to provide a platform for networking and learning.

**Comment by jagannathBhat** [comment]
From the responses that @athul shared in the telegram group the main thing that organizers find difficult is finding a sponsor. What participants are looking for is learning something new and networking. We can't do much about improving the networking part I guess. It's up to the participants. Whereas for learning, we can make sure each hackathon has enough mentors to help the participants. The goal should be as @kaiiyer mentioned, assigning a mentor to each group prior to the event.

So as MVP, let's create a platform focusing mainly on,
- helping organizers find sponsors.
- helping participants find mentors prior to the hackathon.

We can work on other features once these two are done.

**Comment by kaiiyer** [comment]
Yeah so for the sponsors, let's create a curated list containing all the organizations and communities that can offer financial, infrastructure, hardware equipment and mentorship support.
Next let's create a list of the experienced mentors(mostly student devs and industry starters coz managers focus more on business models & can't actually understand the student problems and there's a generation gap).
This list can be like consisting of several sections. Eg. Section 1 WebD and 5 mentors who have sufficient knowledge and projects in webD and can actually mentor devs.
Summing up it'll be an awesome-hackathon list that any organizer can use for making their hackathon *awesome* !!

**Comment by jagannathBhat** [comment]
That's a good solution. However, maintaining a curated list of sponsors sounds centralised and all the work would be on us.

How about we let hackathon organizers create a page for their hackathon on our platform (for example: hhkerala.com/hackcamp2020) and ask them to list their sponsors in that page and how to contact them.

- This shifts the responsibility of maintaining the list from us to our users. It will be like organizers helping each other.
- This helps others find sponsors of hackathons they enjoyed. Recently an organizer wanted to know who sponsored the food for Hack Camp. The organizer forgot the name of the sponsor. But he remembered the hackathon.

Initially we have to request organizers of past events to create pages on our platform.

We also have to provide other usefull features on our platform to encourage organizers to create pages on our platform.

The same pages can be used to create a list of mentors.

**Comment by akash-santhosh** [comment]
Hello All, 
                one thing i've always wanted to Correct in Hackathons is that whenever a hackathon is 36 Hours, it never will be 36 Hours. For example, Take HackCamp, it was intended to Start at 8 AM which delayed two hours, so 10 AM, Hacking should end at 10 PM next Day, instead it stopped at 2 PM.

The Hacking time alone should be 24 or 36 not the whole Event...

**Comment by kaiiyer** [comment]
> Hello All,
> one thing i've always wanted to Correct in Hackathons is that whenever a hackathon is 36 Hours, it never will be 36 Hours. For example, Take HackCamp, it was intended to Start at 8 AM which delayed two hours, so 10 AM, Hacking should end at 10 PM next Day, instead it stopped at 2 PM.
> 
> The Hacking time alone should be 24 or 36 not the whole Event...

I suggest keep the hacking for 24 hours and rest of the events like workshop,registration, results and final pitching in extra hours of the event. Thus making the event a 36 hr hackathon with exactly 24hours for hacking !!

**Comment by jagannathBhat** [comment]
That's a good suggestion. That's something for organizers to keep in mind. However, we are not organizing a hackathon. We are creating tools to help hackers and organizers.

**Comment by akash-santhosh** [comment]
> That's a good suggestion. That's something for organizers to keep in mind. However, we are not organizing a hackathon. We are creating tools to help hackers and organizers.

We Can Suggest Them Right, we are helping to Organize Best Hackathon

**Comment by jagannathBhat** [comment]
Yes, you can suggest that to organizers.

However we are only creating tools to help organizers. It's upto them to conduct the best hackathon or not.

**Comment by athul** [comment]
> That's a good suggestion. That's something for organizers to keep in mind. However, we are not organizing a hackathon. We are creating tools to help hackers and organizers.

We could write a guildelines.md for helping the organizers host wonderful hackathons

**Comment by jagannathBhat** [comment]
Yes. Let's do that.

**Comment by athul** [comment]
https://twitter.com/_phzn/status/1233176546590035968?s=19

This is a super cool thread. We could gain lots from this
